### PR TITLE
Upgrade pnpm to version 9.4.0

### PR DESCRIPTION
**Pull request by KVNLS**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Upgrade pnpm to version 9.4.0

### ❓ Context

- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/browse/LIVE-12963

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: .changeset/beige-needles-develop.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"live-mobile": minor
```
Comment (valpinkman) [inline .changeset/beige-needles-develop.md:2]: not live mobile related, this is a tech task it shouldn't show in changeset

File: .github/workflows/account-migration.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
         uses: LedgerHQ/ledger-live/tools/actions/composites/setup-git-user@develop
 
       - name: Setup the toolchain
-        uses: LedgerHQ/ledger-live/tools/actions/composites/setup-toolchain@develop
+        uses: LedgerHQ/ledger-live/tools/actions/composites/setup-toolchain@fix/upgrade-pnpm
```
Comment (Justkant) [inline .github/workflows/account-migration.yml:34]: do you keep this change for develop too ?

Comment (KVNLS) [reply .github/workflows/account-migration.yml:34]: I put this PR under review too early, sorry. I want to make the CI work with the updated workflows as we pin pnpm for some actions. Then I will drop the commit with this change before merging into develop

**Comment by KVNLS** [comment]
CI is ok, I revert the commit with branch-specific actions

**Pull request merged**